HeterogeneousCore/SonicTriton/src/TritonClient.cc

```diff
@@ -222,30 +267,63 @@ bool TritonClient::handle_exception(F&& call) {
   }
 }
 
-void TritonClient::getResults(std::shared_ptr<tc::InferResult> results) {
-  for (auto& [oname, output] : output_) {
-    //set shape here before output becomes const
-    if (output.variableDims()) {
-      std::vector<int64_t> tmp_shape;
-      TRITON_THROW_IF_ERROR(results->Shape(oname, &tmp_shape), "getResults(): unable to get output shape for " + oname);
-      if (!noBatch_)
-        tmp_shape.erase(tmp_shape.begin());
-      output.setShape(tmp_shape);
-      output.computeSizes();
+void TritonClient::getResults(std::vector<tc::InferResult*>& results) {
+  for (unsigned i = 0; i < results.size(); ++i) {
+    std::shared_ptr<tc::InferResult> result(results[i]);

```

Comment:
Would `unique_ptr` work, or is shared ownership needed?

Reply:
Shared ownership is needed. Multiple `TritonDataEntry` objects can use the same `InferResult` object.

Reply:
Ok (right, the shared ownership is also visible in the for loop below)